p2p/peer.go

```diff
@@ -252,47 +337,74 @@ func (p *peer) SocketAddr() *na.NetAddr {
 	return p.peerConn.socketAddr
 }
 
-// Status returns the peer's ConnectionStatus.
-func (p *peer) Status() tcpconn.ConnectionStatus {
-	return p.mconn.Status()
-}
-
-// Send msg bytes to the channel identified by chID byte. Returns false if the
-// send queue is full after timeout, specified by MConnection.
+// Send sends the given envelope via the stream identified by e.ChannelID. It
+// blocks until the message is sent.
 //
-// thread safe.
-func (p *peer) Send(e Envelope) bool {
-	return p.send(e.ChannelID, e.Message, p.mconn.Send)
+// thread-safe.
+func (p *peer) Send(e Envelope) error {
```

Comment:
I was about to say it is fine until I saw a use case where when the peer is not running the messages is not sent and we'd send a `nil` error. I would define a specific error for non erroneous cases when a message cannot be sent. Because people should know when a message was not sent.  Or leave it with a bool. Or return (bool,error)?

Reply:
I'd say `(bool, error)` is the idiomatic solution in Go for such cases. If it is a recoverable situation (and `Send` on a peer not running sounds like it), I also think we should define an error case for it and return `(false, err)` rather than `panic`ing.

Reply:
> until I saw a use case where when the peer is not running the messages is not sent and we'd send a nil error.

This was an oversight 👍 fixed